aws/sdk/integration-tests/s3/tests/alternative-async-runtime.rs

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_sdk_s3::model::{
+    CompressionType, CsvInput, CsvOutput, ExpressionType, FileHeaderInfo, InputSerialization,
+    OutputSerialization,
+};
+use aws_sdk_s3::{Client, Config, Credentials, Region};
+use aws_smithy_async::assert_elapsed;
+use aws_smithy_async::rt::sleep::{AsyncSleep, Sleep};
+use aws_smithy_client::never::NeverConnector;
+use aws_smithy_http::result::SdkError;
+use aws_smithy_types::timeout;
+use aws_smithy_types::tristate::TriState;
+
+use std::fmt::Debug;
+use std::sync::Arc;
+use std::time::Duration;
+
+#[derive(Debug)]
+struct SmolSleep;
+
+impl AsyncSleep for SmolSleep {
+    fn sleep(&self, duration: Duration) -> Sleep {
+        Sleep::new(async move {
+            smol::Timer::after(duration).await;
+        })
+    }
+}
+
+#[test]
+fn test_smol_runtime_timeouts() {
+    if let Err(err) = smol::block_on(async { timeout_test(Arc::new(SmolSleep)).await }) {
+        println!("{err}");
+        panic!();
+    }
```

Comment:
Simpler to just `unwrap`/`expect` the `block_on` call rather than handling the error and manually panicking.

Reply:
I wanted to print the nice display version of the errors instead of the debug version. I can change it if you don't think that's worth it.

Reply:
`panic!("{err}");`?

It would reduce the lines of code quite a bit without it though 🤷